contracts/models/contract_enterprise.py

```diff
@@ -0,0 +1,50 @@
+from odoo import api, fields, models, tools, _
+from odoo.exceptions import UserError, ValidationError
+from datetime import datetime, timedelta
+
+class ContractEnterprise(models.Model):
+    _name = "contract.enterprise"
+    _description = "contract enterprise model"
+
+    def _default_sign_date(self):
+        next_week = fields.Datetime.now() + timedelta(weeks=1)
+        if next_week.weekday() >= 5:
+            next_week += timedelta(days=7 - next_week.weekday())
+        return next_week
+
+    id = fields.Integer('id', required=True)
+    name = fields.Char("contract name", required=True)
+    start_date = fields.Datetime('start date', required=True, select=True, default=datetime.now())
+    end_date = fields.Datetime('end date', required=True, select=True)
+    type_contract = fields.Integer('contract type', required=True)
+    sign_date = fields.Datetime('sign date', required=True, select=True, default=_default_sign_date)
+    salary_level = fields.Float('salary level', required=False, default=0)
+    effective_salary = fields.Float('effective salary', required=False, default=0)
+    employee_id = fields.Many2one('employee.directory', string="Employee", required=True, ondelete='cascade')
+    '''
+        ondelete : có 4 type
+            + cascade : khi một bản ghi được xóa, tất cả các bản ghi liên quan trong table con sẽ cũng được xóa.
+            + set null :  khi một bản ghi được xóa, trường Many2one sẽ được đặt thành giá trị NULL.
+            + restrict : Odoo sẽ không cho phép xóa bản ghi nếu có bản ghi liên quan trong bảng con.
+            + no action : Odoo sẽ không thực hiện bất kỳ hành động nào khi một bản ghi được xóa. Bạn phải quản lý các hành động liên quan bằng cách thủ công
+    '''
+    sign_day = fields.Date(string="Sign Date")
+    status = fields.Selection([
+                ('new', 'New'),
+                ('running', 'Running'),
+                ('expired', 'Expired'),
+                ('pause', 'Pause')
+            ], string='Status', default='new')
+
+    total_salary = fields.Float(string='Total Salary', compute='_compute_total_salary', store=True)
+
+    @api.depends('salary_level', 'effective_salary')
+    def _compute_total_salary(self):
+        for record in self:
```

Comment:
@Tampham2-ptt Tại sao phải dùng for ở chỗ này?

Reply:
+ sử dụng for và thông quá biến record sẽ an toàn hơn cho những case không biết rõ sẽ có bao nhiêu records được get ra
+ đối với trường hợp này thì có thể không cần dùng for mà có thể dùng trực tiếp biến self để lấy giá trị của record (ví như self.salary_level + self.effective_salary). khi mà chắc chắn sẽ chỉ lấy ra 1 record duy nhất